content/blog/2018-02-07-update-on-async-rendering.md

```diff
@@ -0,0 +1,131 @@
+
+1. As users navigate within an app, newly displayed components often have asynchronous dependencies (including data, images, and code splitting). This leads to a lot of boilerplate code managing data fetching and displaying the loading states. It can also lead to a "cascade of spinners" as the data loads, causing DOM reflows and janky user experience. We'd like to make it easier for product developers to express asynchronous dependencies of components. React could keep the old UI "alive" and interactive for a certain period while the updated UI is not ready yet, and provide a declarative way to show a loading indicator if it takes more than a second.
+2. Fast updates within a short timeframe often cause jank because React processes each update individually. We'd like to automatically "combine" updates within a few hundred milliseconds when possible so that there is less re-rendering.
+3. Some updates are inherently less important than others. For example, if you're writing a live-updating search filter input like [this](https://zeit.co/blog/domains-search-web#asynchronous-rendering), it is essential that the input is updated immediately (within a few milliseconds). Re-rendering the result list can be done later, and should not block the thread or cause stutter when typing. It would be nice if React had a way to mark the latter updates as having a lower priority. (Note that even debouncing the input doesn't help because if the rendering is synchronous—like in React today—a keystroke can't interrupt the rendering once it has started. Asynchronous rendering solves this by splitting rendering into small chunks that can be paused and later resumed.)
+4. For UI elements like hidden popups and tabs, we'd like to be able to start pre-rendering their content when the browser isn't busy. This way, they can appear instantaneously in response to a later user interaction. However, we don't want to make the initial rendering slower, so it's essential to render such elements lazily ([when the browser is idle](https://developers.google.com/web/updates/2015/08/using-requestidlecallback)).
+5. For many apps, React is not the only JavaScript on the page. It often has to coordinate with other JS libraries, server-rendered widgets, and so on. Asynchronous rendering lets React better coordinate with non-React code regarding when components are inserted into the DOM so that [the user experience is smooth](https://twitter.com/acdlite/status/909926793536094209).
+
+Of course, it's possible to implement some of these features today, but it's difficult. We hope to make them effortless by building them into React itself. By replacing problematic lifecycles with safer alternatives, we also hope to make it simple to write async-safe React components.
+
+In the next section, we'll look at how to update your existing components to prepare for the upcoming lifecycle changes.
+
+## Updating class components
```

Comment:
If this is a standalone post (my preference), we will need a section hereabouts summarizing the changes. I would like us to call out that the intention of the change is to make the lifecycle methods more functional and pure. And also say why instance variables are problematic (it's sorta implicit in your post but I'd love to have a sentence people can point to for this saying it's bad).

Reply:
We kind of summarized most of the relevant changes in the intro to this (standalone) post. I think the only thing we haven't explicitly mentioned yet is the new static lifecycle.

I'll try to introduce this. I'm not sure on wording yet.

Reply:
I added a section, "_Before we begin, here’s a quick reminder of the lifecyle changes in version 16.3_". Hopefully this is something like what yo had in mind.